tasks.go

```diff
@@ -137,19 +140,96 @@ func GetAllTasks(runner Runner, basepath string) <-chan Task {
 			defer wg.Done()
 			tasks <- GetOcAdmDiagnosticsTask(runner)
 		}()
+		wg.Wait()
+	}()
+	return tasks
+}
 
+func RunAllAnalysisTasks(runner Runner, path string, workers int) {
+	start := time.Now()
+
+	checkResults := make(chan CheckResults)
+	tasks := GetAllAnalysisTasks(runner, path, checkResults)
+	results := make(chan error)
+
+	// Start worker goroutines to run tasks concurrently.
+	var wg sync.WaitGroup
+	wg.Add(workers)
+	for i := 0; i < workers; i++ {
+		go func() {
+			defer wg.Done()
+			for task := range tasks {
+				results <- task()
+			}
+		}()
+	}
+
+	// Listen to the checkResults channel and write all the results into
+	// the analysis.json file
+	go func() {
```

Comment:
I wonder if something goes wrong here, aren't we risking truncating the file?
IIUC we're overwriting the same file over and over.

The idea seems nice, but perhaps it is safer to write to a different file and move it later? I think `vim` and some other editors and utilities take that approach?